stat/sampleuv/weighted.go

```diff
@@ -54,70 +52,89 @@ func NewWeighted(w []float64, src rand.Source) Weighted {
 // already taken.
 func (s Weighted) Len() int { return len(s.weights) }
 
+func (s Weighted) findIndex(r float64) int {
+	i := 0
+
+	for {
+		r -= s.weights[i]
+		if r < 0 {
+			break // Fall within item i.
+		}
+
+		ni := i<<1 + 1 // Move to left child.
+		// Left node should exist, because r is non-negative, but there are could be floating point errors,
+		// so we check index explicitly.
```

Comment:
```suggestion
		// Left node should exist, because r is non-negative,
		// but there could be floating point errors, so we
		// check index explicitly.
```
(narrow comments are preferred within code)

Reply:
Fixed